mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -753,13 +773,22 @@ else if ( type.getReturnedClass() == ObjectId.class ) {
 		else if ( type instanceof StringAsObjectIdType ) {
 			return StringAsObjectIdGridType.INSTANCE;
 		}
+		else if ( type instanceof BytesAsBsonBinaryType ) {
+			return BytesAsBsonBinaryGridType.INSTANCE;
+		}
+		else if ( type.getReturnedClass() == Long.class && type.getClass() == SerializableToBlobType.class  ) {
+			return LongAsBinaryGridType.INSTANCE;
+		}
```

Comment:
I think what you have to do here is something similar at what I've done for Neo4j with Bolt, check the classes:
`ByteArrayMappedGridTypeDescriptor`
`SerializableAsByteArrayType`

I don't think you should check the returned class and only applied this for Long, I'm pretty sure a user could store other types as well using Lob, it needs to be a  more generic.